fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ConcreteAvailability.java

```diff
@@ -41,13 +44,35 @@ public ConcreteAvailability(
             @NotNull DataSourceMetadataService metadataService
     ) {
         this.name = tableName;
-        this.columns = columns;
         this.metadataService = metadataService;
+        this.names = Collections.singleton(tableName);
+        this.columns = columns;
```

Comment:
Should turn this into an `ImmutableSet` so that callers can't modify it if they get it via `getColumns`